src/query/ovfp_threahold_monitor.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2008 Search Solution Corporation
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * ovfp_threahold_monitor.hpp - (at Server).
+ *
+ */
+#ifndef _OVFP_THRESHOLD_MONITOR_H_
+#define _OVFP_THRESHOLD_MONITOR_H_
+
+#if !defined (SERVER_MODE)
+#error Belongs to server module
+#endif
+
+#if defined (SERVER_MODE)
+
+#include <mutex>
+#include "vacuum.h"
+
+class ovfp_monitor_lock
+{
+#define LOCK_FREE_OWNER_ID (-1)
+#define LOCK_ALL_OWNER_ID  (VACUUM_MAX_WORKER_COUNT)
+#define LOCK_ITEMS_SIZE    (VACUUM_MAX_WORKER_COUNT)
+  private:
+    std::mutex  m_ovfp_monitor_mutex;
+    int         m_lock_arr[LOCK_ITEMS_SIZE];
+
+  public:
+    ovfp_monitor_lock();
+    void lock (int lock_index, int owner_id);
+    void unlock (int lock_index, int owner_id);
+};
```

Comment:
```suggestion
  public:
    ovfp_monitor_lock();
    
    // Deleting the copy constructor and copy assignment operator
    ovfp_monitor_lock(const ovfp_monitor_lock&) = delete;
    ovfp_monitor_lock& operator=(const ovfp_monitor_lock&) = delete;

    // Deleting the move constructor and move assignment operator
    ovfp_monitor_lock(NonCopyableNonMovable&&) = delete;
    ovfp_monitor_lock& operator=(ovfp_monitor_lock&&) = delete;
    
    void lock (int lock_index, int owner_id);
    void unlock (int lock_index, int owner_id);
};
```
Non-copyable and non-movable. If the lock array is copied or moved, It could cause unexpected behavior.

Reply:
I've added the code based on comments.
However, due to the std::mutex originally included in this class, the phenomenon of concern is fundamentally blocked.